tests/test_trajectory.py

```diff
@@ -0,0 +1,305 @@
+
+    survey = resqpy.well.DeviationSurvey(
+        parent_model = model,
+        represented_interp = well_interp,
+        md_datum = datum,
+        **data,
+        **array_data,
+    )
+
+    # --------- Act ----------
+    # Create a trajectory from the deviation survey
+    trajectory_from_deviation_survey = resqpy.well.Trajectory(parent_model = model, deviation_survey = survey)
+
+    # --------- Assert ----------
+    assert trajectory_from_deviation_survey is not None
+    assert trajectory_from_deviation_survey.knot_count == survey.station_count == \
+           trajectory_from_deviation_survey.control_points.shape[0] == 3
+    assert trajectory_from_deviation_survey.md_datum == datum
+    assert trajectory_from_deviation_survey.measured_depths[0] == 1001.0
+
+
+def test_load_from_wellspec(example_model_and_crs):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    grid = RegularGrid(model,
+                       extent_kji = (4, 4, 3),
+                       dxyz = (50.0, -50.0, 50.0),
+                       origin = (0.0, 0.0, 100.0),
+                       crs_uuid = crs.uuid,
+                       set_points_cached = True)
+    grid.write_hdf5()
+    grid.create_xml(write_geometry = True)
+    wellspec_file = os.path.join(model.epc_directory, 'wellspec.dat')
+    well_name = 'DOGLEG'
+    source_df = pd.DataFrame([[2, 2, 1, 0.0, 0.0, 0.0, 0.25], [2, 2, 2, 0.45, -90.0, 2.5, 0.25],
+                              [2, 3, 2, 0.45, -90.0, 1.0, 0.20], [2, 3, 3, 0.0, 0.0, -0.5, 0.20]],
+                             columns = ['IW', 'JW', 'L', 'ANGLV', 'ANGLA', 'SKIN', 'RADW'])
+    with open(wellspec_file, 'w') as fp:
+        fp.write(F'WELLSPEC {well_name}\n')
+        for col in source_df.columns:
+            fp.write(f' {col:>6s}')
+        fp.write('\n')
+        for row_index in range(len(source_df)):
+            row = source_df.iloc[row_index]
+            for col in source_df.columns:
+                if col in ['IW', 'JW', 'L']:
+                    fp.write(f' {int(row[col]):6d}')
+                else:
+                    fp.write(f' {row[col]:6.2f}')
+            fp.write('\n')
+
+    # --------- Act ----------
+    # Create a trajectory from the WELLSPEC file
+    trajectory_from_wellspec = resqpy.well.Trajectory(parent_model = model,
+                                                      well_name = well_name,
+                                                      length_uom = 'm',
+                                                      grid = grid,
+                                                      wellspec_file = wellspec_file)
+
+    # --------- Assert ----------
+    assert trajectory_from_wellspec is not None
+    assert trajectory_from_wellspec.knot_count == len(trajectory_from_wellspec.control_points)
+
+
+def test_load_from_ascii_file(example_model_and_crs):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    well_names = [None, 'Coconut']
```

Comment:
What? No Mango? What's the reason? Is it the wrong season?

Reply:
Sadly there are no mangoes on my Julie mango tree right now #tragictimes